io/gdal/GDALReader.cpp

```diff
@@ -0,0 +1,216 @@
+/******************************************************************************
+* Copyright (c) 2015, Howard Butler <howard@hobu.co>
+*
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following
+* conditions are met:
+*
+*     * Redistributions of source code must retain the above copyright
+*       notice, this list of conditions and the following disclaimer.
+*     * Redistributions in binary form must reproduce the above copyright
+*       notice, this list of conditions and the following disclaimer in
+*       the documentation and/or other materials provided
+*       with the distribution.
+*     * Neither the name of Hobu, Inc. or Flaxen Geo Consulting nor the
+*       names of its contributors may be used to endorse or promote
+*       products derived from this software without specific prior
+*       written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+#include "GDALReader.hpp"
+
+#include <sstream>
+#include <algorithm>
+
+
+#include <pdal/PointView.hpp>
+#include <pdal/GlobalEnvironment.hpp>
+
+namespace pdal
+{
+
+static PluginInfo const s_info = PluginInfo(
+        "readers.gdal",
+        "Read GDAL rasters as point clouds.",
+        "http://pdal.io/stages/reader.gdal.html");
+
+
+CREATE_STATIC_PLUGIN(1, 0, GDALReader, Reader, s_info);
+
+
+std::string GDALReader::getName() const
+{
+    return s_info.name;
+}
+
+
+GDALReader::GDALReader()
+    : m_index(0)
+{}
+
+
+void GDALReader::initialize()
+{
+    GlobalEnvironment::get().initializeGDAL(log());
+    if (!m_raster)
+        m_raster = std::unique_ptr<gdal::Raster>(new gdal::Raster(m_filename));
+
+    m_raster->open();
+    setSpatialReference(m_raster->getSpatialRef());
+    m_count = m_raster->m_raster_x_size * m_raster->m_raster_y_size;
+
+}
+
+void GDALReader::processOptions(const Options& options)
```

Comment:
oh, processOptions. I will be adding more GDAL-specific options in the near future.